features/step_definitions/machine_misc.rb

```diff
@@ -26,4 +26,15 @@
      logger.warn "We will skip this scenario"
      skip_this_scenario
   end
-end 
+end
+
+Given(/^the last provisioned machine is worker$/) do
+  machines = BushSlicer::MachineMachineOpenshiftIo.list(user: admin, project: project("openshift-machine-api"))
+  cache_resources *machines
+  puts machines.max_by(&:created_at).name.include? 'worker'
+  unless (machines.max_by(&:created_at).name.include? 'worker')
+     logger.warn "When no worker machine is present ,workers related cases like workers metrics not valid to test"
```

Comment:
`logger.warn "Skipping this scenario because the last provisioned machine is not worker."`

Reply:
Done